actionpack/lib/action_dispatch/middleware/request_id.rb

```diff
@@ -15,16 +15,15 @@ module ActionDispatch
   # The unique request id can be used to trace a request end-to-end and would typically end up being part of log files
   # from multiple pieces of the stack.
   class RequestId
-    X_REQUEST_ID = "X-Request-Id" #:nodoc:
-
-    def initialize(app)
+    def initialize(app, header: nil)
       @app = app
+      @header = header || "X-Request-Id"
     end
 
     def call(env)
       req = ActionDispatch::Request.new env
-      req.request_id = make_request_id(req.x_request_id)
-      @app.call(env).tap { |_status, headers, _body| headers[X_REQUEST_ID] = req.request_id }
+      req.request_id = make_request_id(req.get_header("HTTP_#{@header.underscore.upcase}"))
```

Comment:
The ENV_METHODS implementation is for hot-path usage. May go ahead with `req.headers[@header]` rather than pull the implementation here.